### ISSUE TITLE
call_transaction_authority_check: false positive

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1812]
false positive in case auth check is done via function module ;-) 

```abap
      CALL FUNCTION 'AUTHORITY_CHECK_TCODE'
        EXPORTING
          tcode  = lv_tcode
        EXCEPTIONS
          ok     = 0
          not_ok = 2
          OTHERS = 3.
      IF sy-subrc = 0.
        CALL TRANSACTION lv_tcode.                     "#EC CI_CALLTA
      ENDIF.
```

**Comment by larshp** [comment]
for this example, I would rewrite the code to
`CALL TRANSACTION lv_tcode WITH AUTHORITY-CHECK.`
instead of calling the function module, less code, easier to read?

**Comment by mbtools** [comment]
Unfortunately, `with authority-check` is a 740 SP 02 feature. 
https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/index.htm?file=abennews-740-others.htm 

Using the function module will be quite common since it's mentioned in the docs, for example 702:
![image](https://user-images.githubusercontent.com/59966492/109722618-ed284d00-7b7a-11eb-9a58-c6a22dd34f60.png)

**Comment by larshp** [comment]
ahh, okay

so it should never give this specific error on 702

and on below 740sp02, do something else, not sure

https://rules.abaplint.org/call_transaction_authority_check/

**Comment by mbtools** [comment]
<740sp2 ignore it,
>=740sp2 keep error but better description with example

Add this function to the rule for replacing functions

**Comment by larshp** [comment]
closing
